modules/repository/generate.go

```diff
@@ -353,3 +371,13 @@ func GenerateRepository(ctx context.Context, doer, owner *user_model.User, templ
 
 	return generateRepo, nil
 }
+
+// Escapes user input to valid OS filenames
+//
+//		Based on https://github.com/sindresorhus/filename-reserved-regex
+//	 Adds "." to prevend directory traversal
+func fileNameEscape(s string) string {
+	re := regexp.MustCompile(`(?i)[\.<>:\"/\\|?*\x{0000}-\x{001F}]|^(con|prn|aux|nul|com\d|lpt\d)$`)
+
+	return strings.TrimSpace(re.ReplaceAllString(s, "_"))
```

Comment:
If it deosn't not allow ".", then "MyFile on 2022.03.04" becomes "MyFile on 2022_03_04" ?

I think a single "." is safe, especially if there is no path separator now.